lib/cn/hblock_builder.c

```diff
@@ -212,19 +234,27 @@ hbb_finish(
     if (num_kblocks == 0 && !wbb_entries(bld->ptree))
         return 0;
 
-    /* Header, HyperLogLog */
-    /* [HSE_TODO]: vblk idx adjust */
-    iov_max = 2;
-    if (wbb_entries(bld->ptree)) {
+    min_pfxp = min_pfxp ? min_pfxp : &min_pfx;
+    max_pfxp = max_pfxp ? max_pfxp : &max_pfx;
+
+    /* Header, vgroup map, and HyperLogLog */
+    iov_max = 3;
+    sz = sizeof(struct vgroup_map_omf) + vgmap->nvgroups * sizeof(struct vgroup_map_entry_omf);
+    vgmap_pgc = roundup(sz, PAGE_SIZE) >> PAGE_SHIFT;
+    assert(vgmap_pgc > 0);
+
+    if (ptree && ptree_pgc > 0) { /* ptree is passed in during kvset split */
```

Comment:
Move the comment above or into the if block

Reply:
I don't see this coming in the way when reading code.

Reply:
To me the comment seems to be describing what the if statement means. Ideally you read the comment prior to the if condition, but this is the opposite.